bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionLocalContext.java

```diff
@@ -59,8 +60,8 @@ private TransactionLocalContext(String globalTransactionId, String url, String p
         this.allowedTransactionRetryCounts = new HashMap<>();
         this.currentTransactionRetryCounts = new HashMap<>();
         this.transactionContextStore = new HashMap<>();
-        this.transactionBlockIdStack = new Stack<>();
-        this.transactionFailure = new Stack<>();
+        this.transactionBlockIdStack = new ConcurrentLinkedDeque<>();
```

Comment:
Shall we use ArrayDeque here since transactionalcontext in bound to same thread.